Severe Climate Events: Disaster Declaration for states and counties

In [2]:
import pandas as pd

In [7]:
disasters_raw = pd.read_csv("disaster_declarations.csv")


In [8]:
disasters_raw.head()

femaDeclarationString  disasterNumber state declarationType  \
0            FM-5444-TX            5444    TX              FM   
1            FM-5436-NE            5436    NE              FM   
2            FM-5444-TX            5444    TX              FM   
3            FM-5436-NE            5436    NE              FM   
4            FM-5436-NE            5436    NE              FM   

            declarationDate  fyDeclared incidentType     declarationTitle  \
0  2022-07-19T00:00:00.000Z        2022         Fire  CHALK MOUNTAIN FIRE   
1  2022-04-23T00:00:00.000Z        2022         Fire        ROAD 702 FIRE   
2  2022-07-19T00:00:00.000Z        2022         Fire  CHALK MOUNTAIN FIRE   
3  2022-04-23T00:00:00.000Z        2022         Fire        ROAD 702 FIRE   
4  2022-04-23T00:00:00.000Z        2022         Fire        ROAD 702 FIRE   

   ihProgramDeclared  iaProgramDeclared  ...  disasterCloseoutDate  \
0                  0                  0  ...                   NaN   
1                  0                  0  ...                   NaN   
2                  0                  0  ...                   NaN   
3                  0                  0  ...                   NaN   
4                  0                  0  ...                   NaN   

   fipsStateCode fipsCountyCode placeCode       designatedArea  \
0             48            221     99221        Hood (County)   
1             31             63     99063    Frontier (County)   
2             48            425     99425   Somervell (County)   
3             31             65     99065      Furnas (County)   
4             31            145     99145  Red Willow (County)   

   declarationRequestNumber  lastIAFilingDate  \
0                     22060               NaN   
1                     22034               NaN   
2                     22060               NaN   
3                     22034               NaN   
4                     22034               NaN   

                                       hash  \
0  373c5ec27998afc08a53302dae796f476b1a6546   
1  ea3487ef36cff455236ce4c63d32fb8b5412bcef   
2  1f35dd8137e1b4cf003fb73d53d8aaaf642e6190   
3  9d813a845ab86f546bdf642fda53d3d4a0fbd098   
4  d436856ce1d5205fa78597b93661fb27d3cea796   

                                     id               lastRefresh  
0  867be42a-71d5-4f13-aa21-d91e0a6fd577  2022-07-20T21:21:23.941Z  
1  e671348b-9782-42df-99f4-d38b8b1a89e6  2022-07-20T21:21:23.942Z  
2  40f3ff75-0b80-4d25-8e85-156cd6a6f40b  2022-07-20T21:21:23.943Z  
3  de47838f-32db-4058-a34c-64997333789e  2022-07-20T21:21:23.943Z  
4  483c03b7-160a-414e-bb4f-5ccecc18d94a  2022-07-20T21:21:23.944Z  

[5 rows x 24 columns]

In [9]:
disasters_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 63989 entries, 0 to 63988
Data columns (total 24 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   femaDeclarationString     63989 non-null  object
 1   disasterNumber            63989 non-null  int64 
 2   state                     63989 non-null  object
 3   declarationType           63989 non-null  object
 4   declarationDate           63989 non-null  object
 5   fyDeclared                63989 non-null  int64 
 6   incidentType              63989 non-null  object
 7   declarationTitle          63989 non-null  object
 8   ihProgramDeclared         63989 non-null  int64 
 9   iaProgramDeclared         63989 non-null  int64 
 10  paProgramDeclared         63989 non-null  int64 
 11  hmProgramDeclared         63989 non-null  int64 
 12  incidentBeginDate         63989 non-null  object
 13  incidentEndDate           55609 non-null  object
 14  disasterCloseoutDate  

In [11]:
disasters = disasters_raw[['id','disasterNumber', 'state','fipsStateCode','fipsCountyCode','placeCode','declarationDate','fyDeclared','incidentType',
    'ihProgramDeclared','iaProgramDeclared','paProgramDeclared','hmProgramDeclared',
    'incidentBeginDate','incidentEndDate','designatedArea']]

Calculating the number of disasters by incident type and removing the ones that are not climate disasters. 

In [18]:
disasters.incidentType.value_counts()
disaster_type_list = disasters.incidentType.unique().tolist()
disaster_type_list
unwanted_disasters = ['Fire', 'Biological','Toxic Substances','Chemical','Terrorist','Human Cause','Fishing Losses']
climate_disasters_list = [ele for ele in disaster_type_list if ele not in unwanted_disasters]
climate_disasters = disasters.loc[disasters['incidentType'].isin(climate_disasters_list)]
climate_disasters

id  disasterNumber state  \
36     036f0c80-8f7d-4551-a29b-92a7923f93a6            1606    TX   
37     d816361f-2c0e-431a-90e5-a836be2ffaf6            1606    TX   
57     a3c1ab84-73dd-4e1b-988f-9968c3a6f005            1971    AL   
114    00c02cd1-ec76-47d1-a5e7-964c4fb6c7b5            4627    DE   
117    3fd860db-e27c-4211-9a69-b314bc68d306            1971    AL   
...                                     ...             ...   ...   
63984  98e40c64-9ade-487f-9a5d-a61a05defd25             136    OR   
63985  6d6279c0-0f40-4b39-9f5c-d1c7a3e83266             368    MS   
63986  5dce9856-d591-471c-bfd2-21738c7f380a             330    MI   
63987  33f53a3e-f779-4acd-b3d5-27862e400a28             297    OK   
63988  b37ffba2-0493-4864-a8fa-f44c376684e9             297    OK   

       fipsStateCode  fipsCountyCode  placeCode           declarationDate  \
36                48              97      99097  2005-09-24T00:00:00.000Z   
37                48             101      99101  2005-09-24T00:00:00.000Z   
57                 1              95      99095  2011-04-28T00:00:00.000Z   
114               10               3      99003  2021-10-24T00:00:00.000Z   
117                1             119      99119  2011-04-28T00:00:00.000Z   
...              ...             ...        ...                       ...   
63984             41               0          0  1962-10-16T00:00:00.000Z   
63985             28             157      99157  1973-03-27T00:00:00.000Z   
63986             26              67      99067  1972-04-05T00:00:00.000Z   
63987             40              19      99019  1970-10-14T00:00:00.000Z   
63988             40             123      99123  1970-10-14T00:00:00.000Z   

       fyDeclared  incidentType  ihProgramDeclared  iaProgramDeclared  \
36           2005     Hurricane                  0                  0   
37           2005     Hurricane                  0                  0   
57           2011  Severe Storm                  1                  0   
114          2022     Hurricane                  0                  0   
117          2011  Severe Storm                  1                  0   
...           ...           ...                ...                ...   
63984        1963  Severe Storm                  0                  1   
63985        1973  Severe Storm                  0                  1   
63986        1972     Snowstorm                  0                  0   
63987        1971  Severe Storm                  0                  1   
63988        1971  Severe Storm                  0                  1   

       paProgramDeclared  hmProgramDeclared         incidentBeginDate  \
36                     1                  1  2005-09-23T00:00:00.000Z   
37                     1                  1  2005-09-23T00:00:00.000Z   
57                     1                  1  2011-04-15T00:00:00.000Z   
114                    1                  1  2021-09-01T00:00:00.000Z   
117                    1                  1  2011-04-15T00:00:00.000Z   
...                  ...                ...                       ...   
63984                  1                  1  1962-10-16T00:00:00.000Z   
63985                  1                  0  1973-03-27T00:00:00.000Z   
63986                  1                  0  1972-04-05T00:00:00.000Z   
63987                  1                  0  1970-10-14T00:00:00.000Z   
63988                  1                  0  1970-10-14T00:00:00.000Z   

                incidentEndDate       designatedArea  
36     2005-10-14T00:00:00.000Z       Cooke (County)  
37     2005-10-14T00:00:00.000Z      Cottle (County)  
57     2011-05-31T00:00:00.000Z    Marshall (County)  
114    2021-09-07T00:00:00.000Z  New Castle (County)  
117    2011-05-31T00:00:00.000Z      Sumter (County)  
...                         ...                  ...  
63984  1962-10-16T00:00:00.000Z            Statewide  
63985  1973-03-27T00:00:00.000Z   Wilkinson (County)  
63986  1972-04-05T00:00:00.000Z       

In [25]:
climate_disasters.dtypes
climate_disasters["incidentBeginDate"] = pd.to_datetime(climate_disasters["incidentBeginDate"])
climate_disasters['Year'] = climate_disasters['incidentBeginDate'].dt.strftime('%Y')


<ipython-input-25-8c89869604a3>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  climate_disasters["incidentBeginDate"] = pd.to_datetime(climate_disasters["incidentBeginDate"])
<ipython-input-25-8c89869604a3>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  climate_disasters['Year'] = climate_disasters['incidentBeginDate'].dt.strftime('%Y')


36       2005
37       2005
57       2011
114      2021
117      2011
         ... 
63984    1962
63985    1973
63986    1972
63987    1970
63988    1970
Name: Year, Length: 52445, dtype: object

In [26]:
climate_disasters.head()

id  disasterNumber state  \
36   036f0c80-8f7d-4551-a29b-92a7923f93a6            1606    TX   
37   d816361f-2c0e-431a-90e5-a836be2ffaf6            1606    TX   
57   a3c1ab84-73dd-4e1b-988f-9968c3a6f005            1971    AL   
114  00c02cd1-ec76-47d1-a5e7-964c4fb6c7b5            4627    DE   
117  3fd860db-e27c-4211-9a69-b314bc68d306            1971    AL   

     fipsStateCode  fipsCountyCode  placeCode           declarationDate  \
36              48              97      99097  2005-09-24T00:00:00.000Z   
37              48             101      99101  2005-09-24T00:00:00.000Z   
57               1              95      99095  2011-04-28T00:00:00.000Z   
114             10               3      99003  2021-10-24T00:00:00.000Z   
117              1             119      99119  2011-04-28T00:00:00.000Z   

     fyDeclared  incidentType  ihProgramDeclared  iaProgramDeclared  \
36         2005     Hurricane                  0                  0   
37         2005     Hurricane                  0                  0   
57         2011  Severe Storm                  1                  0   
114        2022     Hurricane                  0                  0   
117        2011  Severe Storm                  1                  0   

     paProgramDeclared  hmProgramDeclared         incidentBeginDate  \
36                   1                  1 2005-09-23 00:00:00+00:00   
37                   1                  1 2005-09-23 00:00:00+00:00   
57                   1                  1 2011-04-15 00:00:00+00:00   
114                  1                  1 2021-09-01 00:00:00+00:00   
117                  1                  1 2011-04-15 00:00:00+00:00   

              incidentEndDate       designatedArea  Year  
36   2005-10-14T00:00:00.000Z       Cooke (County)  2005  
37   2005-10-14T00:00:00.000Z      Cottle (County)  2005  
57   2011-05-31T00:00:00.000Z    Marshall (County)  2011  
114  2021-09-07T00:00:00.000Z  New Castle (County)  2021  
117  2011-05-31T00:00:00.000Z      Sumter (County)  2011

In [35]:
climate_last_23 = climate_disasters.loc[(climate_disasters['Year'] >= '2000')
                     & (climate_disasters['Year'] <= '2023')]
climate_last_23

id  disasterNumber state  \
36     036f0c80-8f7d-4551-a29b-92a7923f93a6            1606    TX   
37     d816361f-2c0e-431a-90e5-a836be2ffaf6            1606    TX   
57     a3c1ab84-73dd-4e1b-988f-9968c3a6f005            1971    AL   
114    00c02cd1-ec76-47d1-a5e7-964c4fb6c7b5            4627    DE   
117    3fd860db-e27c-4211-9a69-b314bc68d306            1971    AL   
...                                     ...             ...   ...   
58987  e5a603dd-b504-45f9-b30b-cfe7694f1055            1311    GA   
59018  f607f5c6-1f6f-4ff7-a7f0-bda45598c22a            1311    GA   
59019  5a26214d-7749-4d1f-968d-c7f00cd6b72e            1311    GA   
59053  a01f3466-8390-46ed-bc53-5304f1d9f54a            1311    GA   
60548  15e13a58-5b5e-446f-87ed-ac69d68bb7c8            3581    VI   

       fipsStateCode  fipsCountyCode  placeCode           declarationDate  \
36                48              97      99097  2005-09-24T00:00:00.000Z   
37                48             101      99101  2005-09-24T00:00:00.000Z   
57                 1              95      99095  2011-04-28T00:00:00.000Z   
114               10               3      99003  2021-10-24T00:00:00.000Z   
117                1             119      99119  2011-04-28T00:00:00.000Z   
...              ...             ...        ...                       ...   
58987             13             223      99223  2000-01-28T00:00:00.000Z   
59018             13             187      99187  2000-01-28T00:00:00.000Z   
59019             13             311      99311  2000-01-28T00:00:00.000Z   
59053             13             231      99231  2000-01-28T00:00:00.000Z   
60548             78              10      99010  2022-07-25T00:00:00.000Z   

       fyDeclared  incidentType  ihProgramDeclared  iaProgramDeclared  \
36           2005     Hurricane                  0                  0   
37           2005     Hurricane                  0                  0   
57           2011  Severe Storm                  1                  0   
114          2022     Hurricane                  0                  0   
117          2011  Severe Storm                  1                  0   
...           ...           ...                ...                ...   
58987        2000  Severe Storm                  0                  0   
59018        2000  Severe Storm                  0                  0   
59019        2000  Severe Storm                  0                  0   
59053        2000  Severe Storm                  0                  0   
60548        2022         Other                  0                  0   

       paProgramDeclared  hmProgramDeclared         incidentBeginDate  \
36                     1                  1 2005-09-23 00:00:00+00:00   
37                     1                  1 2005-09-23 00:00:00+00:00   
57                     1                  1 2011-04-15 00:00:00+00:00   
114                    1                  1 2021-09-01 00:00:00+00:00   
117                    1                  1 2011-04-15 00:00:00+00:00   
...                  ...                ...                       ...   
58987                  1                  1 2000-01-22 00:00:00+00:00   
59018                  1                  1 2000-01-22 00:00:00+00:00   
59019                  1                  1 2000-01-22 00:00:00+00:00   
59053                  1                  1 2000-01-22 00:00:00+00:00   
60548                  1                  0 2022-07-15 00:00:00+00:00   

                incidentEndDate                          designatedArea  Year  
36     2005-10-14T00:00:00.000Z                          Cooke (County)  2005  
37     2005-10-14T00:00:00.000Z                         Cottle (County)  2005  
57     2011-05-31T00:00:00.000Z                       Marshall (County)  2011  
114    2021-09-07T00:00:00.000Z                     New Castle (County)  2021  
117    2011-05-31T00:00:00.000Z                         Sumter (County)  2011  
...                         ...                      

If we take all years- we have 52,445 climate-related disasters. If we take from 2000 onwards (2000-2023 included), we get 33,629 climate related disasters. 

In [37]:
climate_last_23.state.value_counts()
#climate_last_23.designatedArea.value_counts()

TX    2363
MO    1845
LA    1832
OK    1603
KY    1509
FL    1404
GA    1272
VA    1227
NC    1219
KS    1175
MS    1134
PR    1072
AR     971
AL     971
IA     953
NE     921
NY     895
TN     790
IN     781
WV     713
SC     711
OH     680
ND     675
MN     667
SD     634
PA     600
IL     574
WI     384
CA     372
WA     351
NJ     343
MD     253
MA     240
MI     220
MT     214
ME     212
VT     197
OR     194
NH     193
CO     183
CT     171
NM     150
AK     111
ID     107
AZ      87
NV      75
UT      72
RI      65
HI      45
MP      38
VI      35
WY      32
AS      31
DE      30
DC      17
GU       8
FM       8
Name: state, dtype: int64

For political lens: get the name of the states that have been affected- maybe the highest ones?

For financial lens: counties that are yes and what all was provided to them? 

For socio-economic: counties that were affected- get the demographic breakdown, what are the relationships between the most heavily affected ones and their socio-economic breakdown?

In [ ]:
climate_last_23.to_csv('climate_disasters_2000_2023.csv')